src/EditorFeatures/Core/Implementation/CommentSelection/ToggleLineCommentCommandHandler.cs

```diff
@@ -0,0 +1,191 @@
+            {
+                return ToggleLineComment(commentInfo, selectedSpans);
+            }
+
+            return s_emptyCommentSelectionResult;
+        }
+
+        private CommentSelectionResult ToggleLineComment(CommentSelectionInfo commentInfo,
+            NormalizedSnapshotSpanCollection selectedSpans)
+        {
+            var textChanges = ArrayBuilder<TextChange>.GetInstance();
+            var trackingSpans = ArrayBuilder<CommentTrackingSpan>.GetInstance();
+
+            var linesInSelections = selectedSpans.ToDictionary(
+                span => span,
+                span => GetLinesFromSelectedSpan(span).ToImmutableArray());
+
+            Operation operation;
+            // If all the selections are fully commented, uncomment.
+            if (!linesInSelections.Values.Where(lines => SelectionHasUncommentedLines(lines, commentInfo)).Any())
+            {
+                foreach (var selection in linesInSelections)
+                {
+                    UncommentLines(selection.Value, textChanges, trackingSpans, commentInfo);
+                }
+
+                operation = Operation.Uncomment;
+            }
+            else
+            {
+                foreach (var selection in linesInSelections)
+                {
+                    CommentLines(selection.Key, selection.Value, textChanges, trackingSpans, commentInfo);
+                }
+
+                operation = Operation.Comment;
+            }
+
+            return new CommentSelectionResult(textChanges, trackingSpans, operation);
+        }
+
+        private static void UncommentLines(ImmutableArray<ITextSnapshotLine> commentedLines, ArrayBuilder<TextChange> textChanges,
+            ArrayBuilder<CommentTrackingSpan> trackingSpans, CommentSelectionInfo commentInfo)
+        {
+            foreach (var line in commentedLines)
+            {
+                if (!line.IsEmptyOrWhitespace())
+                {
+                    var text = line.GetText();
+                    var commentIndex = text.IndexOf(commentInfo.SingleLineCommentString) + line.Start;
+                    var spanToRemove = TextSpan.FromBounds(commentIndex, commentIndex + commentInfo.SingleLineCommentString.Length);
+                    DeleteText(textChanges, spanToRemove);
+                }
+            }
+
+            trackingSpans.Add(new CommentTrackingSpan(TextSpan.FromBounds(commentedLines.First().Start, commentedLines.Last().End)));
+        }
+
+        private static void CommentLines(SnapshotSpan selectedSpan, ImmutableArray<ITextSnapshotLine> linesInSelection,
+            ArrayBuilder<TextChange> textChanges, ArrayBuilder<CommentTrackingSpan> trackingSpans, CommentSelectionInfo commentInfo)
+        {
+            var indentation = DetermineSmallestIndent(selectedSpan, linesInSelection.First(), linesInSelection.Last());
+            foreach (var line in linesInSelection)
+            {
+                if (!line.IsEmptyOrWhitespace())
```

Comment:
is this existing behavior? so if I do uncomment

// hello
//
// hello2

and then comment again, I get

// hello

// hello2
?

Reply:
Yes, that's what you'd get, it matches current comment behavior as well as VSCode.